### PR TITLE
fix(ngAnimate): remove prepare classes with multiple structural anima…

### PR DESCRIPTION
**Pull request by Narretz**
…tions

Closes #16677

**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**

**What is the current behavior? (You can also link to an open issue here)**

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change?**

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our [guidelines](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#commits)
- [ ] Fix/Feature: [Docs](https://github.com/angular/angular.js/blob/master/DEVELOPERS.md#documentation) have been added/updated
- [ ] Fix/Feature: Tests have been added; existing tests pass

**Other information**:

### CODE REVIEW

File: src/ngAnimate/animation.js [hunk 1]
```diff
@@ -186,8 +184,9 @@ var $$AnimationProvider = ['$animateProvider', /** @this */ function($animatePro
         forEach(groupedAnimations, function(animationEntry) {
           var element = animationEntry.from ? animationEntry.from.element : animationEntry.element;
           var extraClasses = options.addClass;
+
           extraClasses = (extraClasses ? (extraClasses + ' ') : '') + NG_ANIMATE_CLASSNAME;
-          var cacheKey = $$animateCache.cacheKey(node, event, extraClasses, options.removeClass);
+          var cacheKey = $$animateCache.cacheKey(element[0], animationEntry.event, extraClasses, options.removeClass);
```
Comment (gkalpak) [inline src/ngAnimate/animation.js]: AFAICT, `element` could also be a DOM node (in which case `element[0]` will be undefined), no?

File: test/ngAnimate/integrationSpec.js [hunk 2]
```diff
@@ -468,6 +471,84 @@ describe('ngAnimate integration tests', function() {
       });
     });
 
+
+    it('should remove the prepare classes when different structural animations happen in the same digest', function() {
+      module('ngAnimateMock');
+      inject(function($animate, $compile, $rootScope, $rootElement, $document, $$animateCache) {
+        element = jqLite(
+           // Class animation pn parent element is neeeded so the child elements get the prepare class
```
Comment (gkalpak) [inline test/ngAnimate/integrationSpec.js]: _pn_ --> _on_ (?)

File: test/ngAnimate/integrationSpec.js [hunk 3]
```diff
@@ -468,6 +471,84 @@ describe('ngAnimate integration tests', function() {
...
+            '</div>' +
+          '</div>'
+        );
+
+        $rootElement.append(element);
+        jqLite($document[0].body).append($rootElement);
+
+        $compile(element)($rootScope);
+        // $rootScope.cond = false;
+        // $rootScope.$digest();
```
Comment (gkalpak) [inline test/ngAnimate/integrationSpec.js]: _cough, cough_

File: src/ngAnimate/animation.js (hunk 1 above)
Comment (Narretz) [reply src/ngAnimate/animation.js]: As far as I can see, at this point it's already normalized to be a DOM node

Comment (gkalpak) [reply src/ngAnimate/animation.js]: Yeah, looking at it again, it seems that `element` is treated as a jqLite/jQuery element. So, it's fine.

**Review by gkalpak — APPROVED** [review]

**Comment by swuttich** [comment]
Sorry for answering late (had a week off). Thanks for refining and merging this issue @Narretz.

**Pull request merged**